script.js

```diff
@@ -4,4 +4,229 @@ const mobileMenu = document.querySelector('.mobile-menu');
+  sectionElement.appendChild(gridProject);
+}
+
+Array.from(sectionElement.children).forEach((child, index) => {
+  child.innerHTML = `
+    <div class="multi-post">
+    <h3 class="multi-text-color">${projects[index].description}</h3>
+     <ul class="language display-flex">
+     <li class="language-background">${projects[index].rubySkill}</li>
+     <li class="language-background">${projects[index].cssSkill}</li>
+     <li class="language-background">${projects[index].jsSkill}</li>
+     <li class="language-background">${projects[index].htmlSkill}</li> 
+     </ul>   
+    <button id="${[index]}" class="see-project" type="button">${projects[index].seeProject}</button>
+ </div>
+`;
+});
+
+// modal design begins
+
+//  Display modal on button click
+const seeProjectVar = document.querySelectorAll('.see-project');
+seeProjectVar.forEach((item) => {
+  item.addEventListener('click', () => {
+    const modalContainer = document.createElement('section');
+    modalContainer.className = 'modal-container';
+    modalContainer.id = 'modal-container-id';
+    modalContainer.innerHTML = `
+    <img class="close" src="${projects[item.id].closeButton}" alt="">
+    <img class="mobile-modal" src="${projects[item.id].imageMobile}" alt="">
+    <img class="desktop-modal" src="${projects[item.id].imageDesktop}" alt="">
+ <h2>${projects[item.id].name}</h2>
+ <ul class="modal-skill display-flex">
+  <li class="language-background">${projects[item.id].rubySkill}</li>
+  <li class="language-background">${projects[item.id].cssSkill}</li>
+  <li class="language-background" >${projects[item.id].jsSkill}</li>   
+</ul>
+<p class="mobile-modal-description">${projects[item.id].modalDescription}</p>
+  <div class="modal-btn-container display-flex"><button class="see-project" type="button">See Live</button>
+    <button class="see-project" type="button">See Source</button></div> 
+       
+    `;
+    document.getElementById('modal-main-container-id').appendChild(modalContainer);
+
+    const closeBtn = document.querySelectorAll('.close');
+
+    closeBtn.forEach((el) => {
+      el.addEventListener('click', () => {
+        modalContainer.classList.add('hide');
+      });
+    });
+  });
+});
+
+// form validation
+
+const email = document.querySelector('#email');
+
+// local storage
+
+const userName = document.querySelector('#full-name');
+const userMsg = document.querySelector('#text-area');
+
+function dataStorage() {
+  const visitor = {
+    Name: userName.value,
+    Email: email.value,
+    Message: userMsg.value,
+  };
+  localStorage.setItem('visitor', JSON.stringify(visitor));
+}
+
+userName.addEventListener('focusout', dataStorage);
+email.addEventListener('focusout', dataStorage);
+userMsg.addEventListener('focusout', dataStorage);
+
+const userDataObject = JSON.parse(localStorage.getItem('visitor'));
+
+if (userDataObject) {
+  userName.value = userDataObject.Name;
+  email.value = userDataObject.Email;
+  userMsg.value = userDataObject.Message;
+}
```

Comment:
**[REQUIRED CHANGES]**

- [ ] Hey, your form works pretty fine but has one issue. After filling out the form, and not submitting it, the entered values should be saved in the browser. The full name and email get refilled but the message doesn't get refilled but only gets refilled after making the form submission. Please, kindly fix this so that all inputs are auto-filled without making any submission to your form.

Reply:
**[OPTIONAL]**

- Please, kindly remove the unnecessary newline added inside your code block. e:g in your `project 6, 5` etc. as the closing curly braces specify the end of your code block, therefore, adding the new line is not ideal and doesn't make your code base look great. :cool: